pkgs/applications/virtualization/anbox/default.nix

```diff
@@ -0,0 +1,125 @@
+      rm -rf $out/include/{gmock,gtest} $out/lib/lib{gmock,gtest}{,_main}.a
+    '';
+  };
+
+  dbus-cpp = stdenv.mkDerivation rec {
+    name = "dbus-cpp-${rev}";
+    rev = "5.0.0+17.04.20170406-0ubuntu1";
+    src = fetchbzr {
+      url = "http://bazaar.launchpad.net/~phablet-team/dbus-cpp/trunk";
+      inherit rev;
+      sha256 = "0mrqh6ynj41fscrhzgagd8m9s45flvrqdacbhk8bi1337vy68m9w";
+    };
+    nativeBuildInputs = [ cmake pkgconfig ];
+    propagatedBuildInputs = [ process-cpp libxml2 dbus ];
+    buildInputs = [ gtest ];
+    preConfigure = ''
+      substituteInPlace CMakeLists.txt --replace 'include(cmake/PrePush.cmake)' "" # skip making .deb
+    '';
+    cmakeFlags = "-DDBUS_CPP_VERSION_MAJOR=5 -DDBUS_CPP_VERSION_MINOR=0 -DDBUS_CPP_VERSION_PATCH=0 -DGMOCK_SOURCE_DIR=${gmock.src}";
+    NIX_CFLAGS_COMPILE = "-Wno-error";
+    postInstall = ''
+      rm -rf $out/include/{gmock,gtest} $out/lib/lib{gmock,gtest}{,_main}.a $out/libexec
+    '';
+  };
+
+  rev = "76be0e2";
+  version = "0.1.0-${rev}";
+  src = fetchFromGitHub {
+    owner = "anbox";
+    repo = "anbox";
+    inherit rev;
+    sha256 = "049wyfn147y97sr53filkfmy5v7873671fvzjh98n90snnlw0h7x";
+  };
+  meta = with stdenv.lib; {
+    homepage = "http://anbox.io";
+    description = "Container based approach to boot a full Android system";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.volth ];
+  };
+in {
+  ashmem = stdenv.mkDerivation {
+    name = "anbox-ashmem-${version}";
+    inherit src meta;
+    sourceRoot = "anbox-${rev}-src/kernel/ashmem";
+    hardeningDisable = [ "pic" ];
+    makeFlags = "KERNEL_SRC=${linuxPackages.kernel.dev}/lib/modules/${linuxPackages.kernel.modDirVersion}/build DESTDIR=$(out)";
+  };
+
+  binder = stdenv.mkDerivation {
+    name = "anbox-binder-${version}";
+    inherit src meta;
+    sourceRoot = "anbox-${rev}-src/kernel/binder";
+    hardeningDisable = [ "pic" ];
+    makeFlags = "KERNEL_SRC=${linuxPackages.kernel.dev}/lib/modules/${linuxPackages.kernel.modDirVersion}/build DESTDIR=$(out)";
+  };
+
+  exe = stdenv.mkDerivation {
+    name = "anbox-${version}";
+    inherit src meta;
+  
+    nativeBuildInputs = [ cmake pkgconfig ];
+    buildInputs = [ boost SDL2 mesa_noglu glm protobuf dbus-cpp lxc gtest xorg.libpthreadstubs xorg.libXdmcp ];
+  
+    preConfigure = ''
+      substituteInPlace CMakeLists.txt --replace 'add_subdirectory(tests)' "" # skip tests, they unable to find libgtest.a
+    '';
+    cmakeFlags = "-DCMAKE_INSTALL_LIBDIR=lib";
+  
+    postInstall = ''
+      cat > $out/bin/runme <<EOF
+      #!${bash}/bin/bash
+
+      $out/bin/anbox session-manager &
+      ${coreutils}/bin/sleep 2
+
+      $out/bin/anbox launch --package=org.anbox.appmgr --component=org.anbox.appmgr.AppViewActivity
+      EOF
+      chmod +x $out/bin/runme
+    '';
+  };
+
+  image = fetchurl {
+    inherit meta;
+    url = "http://build.anbox.io/android-images/2017/04/12/android_1_amd64.img";
```

Comment:
Building an android image takes a lot of resources and the build system is not yet prepared for nix, so downloading a precompiled image is probably ok at the moment.